src/main/resources/static/admin.html

```diff
@@ -1,10 +1,62 @@
 <!DOCTYPE html>
 <html lang="en">
 <head>
+    <title>Minimal Search Engine</title>
     <meta charset="UTF-8">
-    <title>Title</title>
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="style.css">
+    <link rel="icon" href="logos/logo_2.png" type="image/x-icon">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Kameron&display=swap" rel="stylesheet">
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/css/bootstrap.min.css">
+    <script type="text/javascript">
+        window.onload = function () {
+            let firstName = localStorage.getItem("adminFirstName")
+            let lastName = localStorage.getItem("adminLastName")
+            let username = localStorage.getItem("adminUsername")
+
+            if (firstName == null || lastName == null || username == null) {
+                alert("Please login before trying to perform an administrative task. Redirecting to login ... ")
+                window.location.href = window.location.origin + "/login.html";
+            }
+            else {
+                $("#adminInfo")
```

Comment:
This may be outside the scope of the project with regards to security, but since it seems the only check for "an admin is logged in" is whether or not these three variables have values, is it possible that a malicious actor could add values for these variables via the console and therefore be "logged in" without entering a password?

Reply:
as you mentioned it is out of scope. On the side, even if someone tries to add the name via console, one still has to brute force and try a possibly infinite combinations to get the username correct to make the admin api work.

Reply:
Sounds good. I think the more robust ways to store and detect an active session would be way too much work and are not necessary.